File: dotnet/src/Buble.Sdk/WaitOptions.cs
namespace Buble.Sdk;

/// <summary>
/// Polling options for asynchronous generation tasks.
/// </summary>
public sealed class WaitOptions
{
    public TimeSpan Interval { get; set; } = TimeSpan.FromSeconds(2);

    public TimeSpan Timeout { get; set; } = TimeSpan.FromMinutes(10);

    public static WaitOptions Default => new();
}

File: swift/Sources/Buble/WaitOptions.swift
/// Polling options for generation wait helpers.
public struct WaitOptions: Sendable {
    public var interval: TimeInterval
    public var timeout: TimeInterval
    public var throwOnFailed: Bool
    public var throwOnCanceled: Bool

    public init(
        interval: TimeInterval = 2,
        timeout: TimeInterval = 600,
        throwOnFailed: Bool = true,
        throwOnCanceled: Bool = true
    ) {
        self.interval = interval
        self.timeout = timeout
        self.throwOnFailed = throwOnFailed
        self.throwOnCanceled = throwOnCanceled
    }
}
